### PR TITLE
qt6: Update to 6.5.0 including portfile clean up

### PR DESCRIPTION
**Pull request by jhoyt4**
qt6: Update to 6.5.0 including portfile clean up and new qt modules     
Updates to qt6 Portfile for 6.5.0

**Note 1:** no-rpath was removed to get qtdeclarative to build
**Note 2:** qtwebengine qtwebview are held back as a protobuff3-cpp is out of date and a newer version is required.

- Make new qt62 port from qt6 Portfile for macOS < 11 now that qt6.5    
  drops support for those OSes                                          
- Update minimum macOS versions for qt6 per qt6.5 documentation         
- Update patches for qt6.5/remove deprecated macOs 10.14 patch (qt6.5)  
- Remove qtbase paths from configure args - now handled automatically   
  via configure (and cause cmake file issues)                           
- Remove qt-base -no-rpath flag (causes issues with qt-declarative)     
- Add macports frameworks dir to qtbase build search path               
- Remove unnecessary cmake file mods (carryover from the qt5 portfile)  
- Remove now unnecessary rglob function                                 
- Add new Subports: qtdatavis3d qtgrpc qtlanguageserver qtlocation      
          qtquick3d qtquick3dphysics qtquickeffectmake qtquicktimeline  
          qtquicktimeline qtquicktimeline qtscxml qtspeech              
          qtvirtualkeyboard qthttpserver       
- Add vulkan support                                                    
- Add dummy cmake find modules for applications built outside of macports
                                                                        
Updates for now unsupported macOS versions                              
- Make new qt62 port from qt6 Portfile for macOS < 11 now that qt6.5    
- Update qt6     

Closes: https://trac.macports.org/ticket/66975

#### Description
Updates QT 6 to 6.5.
Copies old qt6 Portfile/patched to qt62 for now unsupported versions of macOS (<11)

###### Type(s)

- [ ] bugfix
- [x] enhancement
- [ ] security fix

###### Tested on

macOS 13.3 22E252 x86_64
Xcode 14.3 14E222b

macOS 13.3 22E252 arm64
Xcode 14.3 14E222b

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [x] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL? 
- [ ] checked your Portfile with `port lint --nitpick`?
- [x] tried existing tests with `sudo port test`?
- [x] tried a full install with `sudo port -vst install`?
- [ ] tested basic functionality of all binary files?
- [x] checked that the Portfile's most important [variants](https://trac.macports.org/wiki/Variants) haven't been broken?

### CODE REVIEW

**Comment by jhoyt4** [comment]
This PR is a WIP.  Still todo:
- Verify at least qtbase with a test application
- Handle qt6 for now unsupported macOS versions (10.14 and 10.15)
- Add missing qt submodules
- Re-verify all required ports
- Possibly add in universal build logic from [PR 16735](https://github.com/macports/macports-ports/pull/16735)

File: _resources/port1.0/group/qt6_version_info-1.0.tcl [hunk 1]
```diff
@@ -5,7 +5,7 @@
 
 global available_qt_versions
 array set available_qt_versions {
-    qt6 {qt6-qtbase 6.2}
+    qt6 {qt6-qtbase 6.5}
```
Comment (aeiouaeiouaeiouaeiouaeiouaeiou) [inline _resources/port1.0/group/qt6_version_info-1.0.tcl:8]: Why can't we make Qt 6.5 a separate version like in the [qt5_version_info](https://github.com/macports/macports-ports/blob/1719a40fce26c09ed29c4ecda3fca6b2fd1bc7a7/_resources/port1.0/group/qt5_version_info-1.0.tcl) portgroup?

**Review by aeiouaeiouaeiouaeiouaeiouaeiou — CHANGES_REQUESTED** [review]

File: _resources/port1.0/group/qt6_version_info-1.0.tcl (hunk 1 above)
Comment (jhoyt4) [reply _resources/port1.0/group/qt6_version_info-1.0.tcl:8]: That is the plan.  This was just a placeholder for me to keep moving forward while that path forward gets sorted.

Here's the thread on the [dev list](https://lists.macports.org/pipermail/macports-dev/2023-April/044984.html)

**Comment by jhoyt4** [comment]
That last force push added all of the missing qt6 submodules except for qt-doc.  Very surprised I was able to get them working this time.  Removing the no-rpath flag seems to have fixed quite a few things.

**Comment by jhoyt4** [comment]
@TellowKrinkle - I've attempted to replicate your suggested dependency updates with qt6.5.0.  Could you provide me a sanity check to make sure I got them all?

**Comment by TellowKrinkle** [comment]
I haven't tested anything, but everything looks about right

**Comment by jhoyt4** [comment]
Interesting note, QtDeclarative and QtWebEngine do not like the -no-rpath flag on ARM.  Tested with and without the related "-R" and "-libdir" flags with no success.  Curiously, -no-rpath works for QtDeclarative on intel...

**Comment by jhoyt4** [comment]
This is close to being ready.  Currently I'm struggling with qt6-qtwebengine and protobuffers.  They're out of date in the ports tree

```
#error This file was generated by a newer version of protoc which is
 ^
gen/components/enterprise/common/proto/download_item_reroute_info.pb.h:13:2: error: incompatible with your Protocol Buffer headers. Please update
#error incompatible with your Protocol Buffer headers. Please update
 ^
gen/components/enterprise/common/proto/download_item_reroute_info.pb.h:14:2: error: your headers.
#error your headers.
```

File: _resources/port1.0/group/qt6_version_info-1.0.tcl (hunk 1 above)
Comment (jhoyt4) [reply _resources/port1.0/group/qt6_version_info-1.0.tcl]: I've just pushed the logic to allow macOS version < 11 to build 6.2 instead

**Comment by jhoyt4** [comment]
I've held back qtwebengine and qtwebview.

qtwebengine is failing to compile due to an old version or protoc provided by protobuff3-cpp.  This needs to be updated, but unfortunately is more than a simple rev update as it now needs bazel to build (which is also dated in the ports tree).

qtwebview is dependent on qtwebengine

Here is the protoc related build error for qtwebenginer:
>
> ```
> #error This file was generated by a newer version of protoc which is
>  ^
> gen/components/enterprise/common/proto/download_item_reroute_info.pb.h:13:2: error: incompatible with your Protocol Buffer headers. Please update
> #error incompatible with your Protocol Buffer headers. Please update
>  ^
> gen/components/enterprise/common/proto/download_item_reroute_info.pb.h:14:2: error: your headers.
> #error your headers.
> ```

**Comment by jhoyt4** [comment]
Converting back to a draft now that the major refactor has been pushed

**Comment by tuffnatty** [comment]
@jhoyt4 If you are still working on this: We on Mojave and Catalina need qt62 port badly as 6.4 cannot actually be built with old XCode which is compatible with Mojave and Catalina.

**Comment by tuffnatty** [comment]
Thank you @tenzap . No, this problem does not affect me. I did not try qtmultimedia but these:
MacBook-Pro% ls -lR /opt/local/var/macports/software/qt6-*|grep 6.2.4
-rw-r--r--  1 macports  wheel  358334 Mar 19 11:53 qt6-qt5compat-6.2.4_2.darwin_18.x86_64.tbz2
-rw-r--r--  1 macports  wheel  12558132 Mar 16 22:38 qt6-qtbase-6.2.4_2+openssl.darwin_18.x86_64.tbz2
-rw-r--r--  1 macports  wheel  12753387 Mar 19 11:43 qt6-qtdeclarative-6.2.4_2.darwin_18.x86_64.tbz2
-rw-r--r--  1 macports  wheel  84132 Mar 19 11:42 qt6-qtimageformats-6.2.4_2.darwin_18.x86_64.tbz2
-rw-r--r--  1 macports  wheel  1201546 Mar 19 11:43 qt6-qtshadertools-6.2.4_2.darwin_18.x86_64.tbz2
-rw-r--r--  1 macports  wheel  182821 Mar 19 11:43 qt6-qtsvg-6.2.4_2.darwin_18.x86_64.tbz2
-rw-r--r--  1 macports  wheel  6673397 Mar 19 15:19 qt6-qttools-6.2.4_2.darwin_18.x86_64.tbz2
-rw-r--r--  1 macports  wheel  559982 Mar 16 22:38 qt6-sqlite-plugin-6.2.4_0.darwin_18.x86_64.tbz2

were built on my machine successfully. After the 6.4 update, clang (from XCode 11.3) is suddenly too old to build qcompilerdetection.h. The same effect is seen on MacPorts binary packages buildbot: https://packages.macports.org/qt6-qtbase/ . Both versions mention being runtime-compatible with 10.14+, but requiring XCode 13+ (SDK 12+) for building. But 6.2 was actually working with XCode 11.3, and 6.4 is not.

**Comment by jhoyt4** [comment]
> @jhoyt4 If you are still working on this.

@tuffnatty - Sorry but no, hence why I closed it.  The portfile does work, but is based on the older method before a major refactor.  You may want to ping the port maintainer for 6.2 as 6.4 and 6.5 do not support older macOS versions.

**Pull request closed without merging**